File: Lab/SEM4/MPP/lab9/moto_c/src/Common/repository/DatabaseConnection.cs
using MySqlConnector;
using Microsoft.Extensions.Configuration;

public class DatabaseConnection
{
    private static DatabaseConnection? instance;
    private readonly string connectionString;

    private DatabaseConnection()
    {
        var config = new ConfigurationBuilder()
            .AddJsonFile("appsettings.json")
            .Build();

        string url = config["Database:Url"]!;
        string user = config["Database:Username"]!;
        string password = config["Database:Password"]!;

        connectionString = $"{url}User={user};Password={password};";
    }

    public static DatabaseConnection getInstance()
    {
        if (instance == null)
            instance = new DatabaseConnection();
        return instance;
    }

    public MySqlConnection GetConnection()
    {
        var connection = new MySqlConnection(connectionString);
        connection.Open();
        return connection;
    }
}

File: Lab/SEM4/MPP/lab4/moto_java/src/main/java/org/zdroba/repository/DataBaseConnection.java
package org.zdroba.repository;

import java.io.IOException;
import java.io.InputStream;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.util.Properties;

public class DataBaseConnection {

    private static DataBaseConnection instance;
    private final Connection connection;

    private DataBaseConnection() {
        Properties props = new Properties();

        try (InputStream input = getClass().getClassLoader().getResourceAsStream("db.config")) {

            if (input == null)
                throw new RuntimeException("db.config file not found");

            props.load(input);
            String url = props.getProperty("jdbc.url");
            String user = props.getProperty("jdbc.username");
            String password = props.getProperty("jdbc.password");

            this.connection = DriverManager.getConnection(url, user, password);
        } catch (IOException | SQLException e) {
            throw new RuntimeException("Failed to initialize database connection", e);
        }
    }

    public static DataBaseConnection getInstance() {
        if (instance == null)
            instance = new DataBaseConnection();
        return instance;
    }

    public Connection getConnection() {
        return connection;
    }
}